File: ProjectApi/ProjectApi/Services/PaintingService.cs
using ProjectApi.Data.Models;
using ProjectApi.Data.Repositories;

namespace ProjectApi.Services
{
    public class PaintingService : IPaintingService
    {
        private readonly PaintingRepository _repository;

        public PaintingService(PaintingRepository repository)
        {
            _repository = repository;
        }

        public async Task<Painting> CreatePainting(Painting painting)
        {
            return await _repository.CreatePainting(painting);
        }

        public async Task DeletePainting(Guid paintingId)
        {
            await _repository.DeletePainting(paintingId);
        }

        public async Task<List<Painting>> GetAllPaintings()
        {
            var paintings = await _repository.GetAllPaintings();
            return paintings;
        }

        public async Task<Painting?> GetPainting(Guid paintingId)
        {
            return await _repository.GetPainting(paintingId);
        }

        public async Task<List<Painting>> GetPaintings(List<Guid> paintingIds)
        {
            return await _repository.GetPaintings(paintingIds);
        }

        public async Task<Painting> UpdatePainting(Painting painting)
        {
            return await _repository.UpdatePainting(painting);
        }
    }
}

File: project-client/src/api/paintingService.ts

import axiosInstance from '../api/axiosInstance';
import { Painting } from '../models/Painting';

export const getPainting = async (id: string): Promise<Painting> => {
    const response = await axiosInstance.get<Painting>(`/GetPainting?id=${id}`);
    return response.data;
};

export const getAllPaintings = async (): Promise<Painting[]> => {
    const response = await axiosInstance.get<Painting[]>('/GetAllPaintings');
    return response.data;
};

export const getPaintings = async (paintingIds: string[]): Promise<Painting[]> => {
    const response = await axiosInstance.post<Painting[]>('/GetPaintings', { paintingIds });
    return response.data;
};

export const createPainting = async (painting: Painting): Promise<Painting> => {
    const response = await axiosInstance.post<Painting>('/CreatePainting', painting);
    return response.data;
};

export const updatePainting = async (painting: Painting): Promise<void> => {
    await axiosInstance.put('/UpdatePainting', painting);
};

export const deletePainting = async (paintingId: string): Promise<void> => {
    await axiosInstance.delete(`/DeletePainting?paintingId=${paintingId}`);
};
